service/lib/agama/storage/proposal_strategies/base.rb

```diff
@@ -72,6 +72,8 @@ def disk_analyzer
 
         # Available devices for installation.
         #
+        # @deprecated Only used by Guided strategy.
+        #
```

Comment:
Why don't we move the method to the Guided class then?

Reply:
In general. I think it makes sense to change `ProposalStrategies::Base` to work with the new class system, instead of only implementing that change for the Agama strategy.

Or there is anything stopping us from doing so?

Reply:
Yes, I think we can use `Storage::System`.